File: c/claude/p9.c
// Task p9: Command line argument indexing
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
    if (argc > 1)
    {
        int index = atoi(argv[1]); // First argument after program name

        // argv[0] is the program name
        // argv[1] is the first user argument (index we want)
        // argv[index + 1] is the argument at the specified index

        if (index > 0 && index < argc)
        {
            printf("Argument at user index %d: %s\n", index - 1, argv[index]);
        }
        else
        {
            printf("Index out of range\n");
        }
    }
    else
    {
        printf("Usage: %s <index>\n", argv[0]);
        printf("Example: %s 1\n", argv[0]);
    }

    return 0;
}

File: javascript/claude/p9.js
// Task p9: Command line argument indexing
const args = process.argv.slice(2); // Get user arguments (skip node and script)

if (args.length > 0) {
  const index = parseInt(args[0]); // First argument after program name

  // args[0] is the first user argument (index we want)
  // args[index] is the argument at the specified index + 1

  if (!isNaN(index) && index >= 0 && index < args.length) {
    console.log(`Argument at user index ${index}: ${args[index]}`);
  } else {
    console.log("Index out of range");
  }
} else {
  console.log("Usage: node p9.js <index>");
  console.log("Example: node p9.js 1");
}
